libbeat/outputs/elasticsearch/elasticsearch.go

```diff
@@ -177,20 +178,16 @@ func makeES(
 }
 
 func buildSelectors(
+	im outputs.IndexManager,
 	beat beat.Info,
 	cfg *common.Config,
-) (index outil.Selector, pipeline *outil.Selector, err error) {
+) (index outputs.IndexSelector, pipeline *outil.Selector, err error) {
 	if !cfg.HasField("index") {
 		pattern := fmt.Sprintf("%v-%v-%%{+yyyy.MM.dd}", beat.IndexPrefix, beat.Version)
 		cfg.SetString("index", -1, pattern)
 	}
```

Comment:
Let me find a way. This is some legacy I'd like to be removed/reduced in general. One module/packge must not use any `cfg.SetX` to change the configuration objects on behalve of another modules. The config object is shared and it is not clear, when parts of it are parsed or not. Every use of `cfg.SetX` makes some implicit assumptions about execution order and how configs are shared when. Which is not always correct thanks to config reloading via auto-discovery or central management. The uses of `cfg.SetX` is what made the process of refactoring the index managenemt quite delicate.

Reply:
I agree, that's a tricky part and non trivial. 
Kind of related, the index supporter generally could mess with the raw config object. However, if there were any changes to the output config, they would not affect the `b.Beat.Config.Output`, as this is set earlier already, see https://github.com/elastic/beats/pull/10347/files#diff-f7226280e7734d6e3d88e7dde232602eR631. (I'm not fully clear yet if this could lead to issues).

Reply:
done